File: app/src/main/java/ua/te/hackathon/smartcity2015/ui/base/mvp/Presenter.java
package ua.te.hackathon.smartcity2015.ui.base.mvp;

/**
 * @author victor
 * @since 2016-02-13
 */
public interface Presenter<T extends PresenterView> {

  void attachView(T view);

  void detachView();

  void onDestroy();

}

File: app/src/main/java/ua/te/hackathon/smartcity2015/ui/base/mvp/Presenter.kt
package ua.te.hackathon.smartcity2015.ui.base.mvp

/**
 * @author victor
 * *
 * @since 2016-02-13
 */
interface Presenter<T : PresenterView> {

  fun attachView(view: T)

  fun detachView()

  fun onDestroy()

}